lib/ace/ext/papeeria/katex-previewer.coffee

```diff
@@ -7,25 +7,21 @@ define((require, exports, module) ->
         placement: "bottom"
         trigger: "manual"
         title: "Formula"
-        container: "#editor"
+        container: editor.container
       }
 
-      show: (jqPopoverContainer, content) ->
-        cursorPosition = $("textarea.ace_text-input").position()
-        jqPopoverContainer.css({
-          top: "#{cursorPosition.top + 24}px"
-          left: "#{cursorPosition.left}px"
-        })
+      show: (jqPopoverContainer, content, position) ->
+        jqPopoverContainer.css(position)
         popoverHandler.options.content = content
         jqPopoverContainer.popover(popoverHandler.options)
         jqPopoverContainer.popover("show")
         return
 
-      hide: (jqPopoverContainer) ->
+      destroy: (jqPopoverContainer) ->
         jqPopoverContainer.popover("destroy")
 
-      isVisible: (jqPopoverElement) ->
-        jqPopoverElement.children(".popover").is(":visible")
+      isVisible: (jqPopoverContainer) ->
+        jqPopoverContainer.data().popover.tip().hasClass("in")
```

Comment:
However, this code looks more fragile and less commonly used. It is fragile because relies on existence of `data().popover.tip()` and some particular class. Should anything break in this chain, or should class name change (e.g. with the new version of Bootstrap), this code is broken. 

With respect to conventional practices, one may expect that visibility of a big element is checked with the test of property "visible", and when a programmer sees this code, he just trusts that it produces the expected result. The same result, produced by checking the class of some particular small element is more difficult to trust -- the reader really gets confused and tries to grasp why this very way was used instead of more conventional one.